src/components/Pill/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+
+const getPills = (statuses) => {
+  const pills = [];
+  statuses.forEach((status) => {
+    switch (status) {
+      case 'unpublished':
+        pills.push({
+          text: 'Incomplete',
+          className: 'badge badge-danger',
+        });
+        break;
+      case 'reviewed':
+        pills.push({
+          text: 'Scheduled',
+          className: 'badge badge-primary',
+        });
+        break;
+      case 'review_by_legal':
+      case 'review_by_internal':
+        pills.push({
+          text: 'In review',
+          className: 'badge badge-warning',
+        });
+        pills.push({
+          text: <i className="fa fa-lock" />,
+        });
+        break;
+      case 'published':
+        pills.push({
+          text: 'Live',
+          className: 'badge badge-success',
+        });
+        break;
+      default:
+        break;
+    }
+  });
+  return pills.map(pill => (
+    <span className={classNames('ml-2', pill.className)}>
+      {pill.text}
+    </span>
+  ));
+};
+
+const Pill = ({
+  statuses,
+}) => getPills(statuses);
+
+Pill.propTypes = {
+  statuses: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.bool, PropTypes.string])).isRequired,
```

Comment:
Why allow bool here? Doesn't look like that case is handled in `render`?

Reply:
It allows bool here because `courseStatuses` come in as either one of the switch cases or False if the condition to show it is not true.

Reply:
Ah, I see now. Why not just have those switches do something like:?
```
if (condition)
   push(status)
```

Reply:
Because the conditions are only checked as part of adding a pill to a course because we have to check what the status of the course runs are.
So the switches are only looking at the values of possible CourseRun statuses, but https://github.com/edx/publisher-frontend/pull/121/files#diff-fa7269192ac8344c05273e80552d7c0eR161 is setting which statuses should be applied to the course by doing multiple checks and if the check is not valid, it returns `false`

Reply:
Updated to be if statements.